pkg/storegateway/bucket.go

```diff
@@ -1798,10 +1731,17 @@ func newBucketBlockSet() *bucketBlockSet {
 	return &bucketBlockSet{}
 }
 
-func (s *bucketBlockSet) add(b *bucketBlock) error {
+// add adds a block to the set. If the block with the same id was already present in the set, add ignores it.
+func (s *bucketBlockSet) add(b *bucketBlock) {
```

Comment:
isn't it a bug to add the same block twice? If we ignore this error, then it might hard to catch bugs.

Reply:
in reality adding a block happens in a single goroutine, so this isn't a problem, but probably better be explicit about these bugs. I'd say the same for `remove`

Reply:
Since this shouldn't happen anyway, shall we panic if _somehow_ we ended adding the same block to `bucketBlockSet`?

Reply:
panic looks a git more brutal. I think a log line is fine. Returning an error should result in a log line